lib/streamlit/web/server/oauth_authlib_routes.py

```diff
@@ -130,13 +131,57 @@ def _parse_provider_token(self) -> str | None:
 class AuthLogoutHandler(AuthHandlerMixin, tornado.web.RequestHandler):
     def get(self) -> None:
         self.clear_auth_cookie()
-        self.redirect_to_base()
+
+        provider_logout_url = self._get_provider_logout_url()
+        if provider_logout_url:
+            self.redirect(provider_logout_url)
+        else:
+            self.redirect_to_base()
+
+    def _get_provider_logout_url(self) -> str | None:
+        """Get the OAuth provider's logout URL from OIDC metadata."""
+        try:
+            cookie_value = self.get_signed_cookie(AUTH_COOKIE_NAME)
+        except AttributeError:  # Backward compatibility with Tornado < 6.3.0
+            cookie_value = self.get_secure_cookie(AUTH_COOKIE_NAME)
```

Comment:
@vdonato this is the workaround for version incompatibility. 
It is modelled after a similar workaround in browser_websocket_handler.py